explainer.md

```diff
@@ -92,33 +90,58 @@ The following code examples try to clarify the proposed IDL API.
 ### Adding anchors
 
 ```javascript
-var anchorToModelMap = new Map();
-// Create an arbitrary anchor
-frame.addAnchor(anchorPose, eyeLevelReferenceSpace).then((anchor) => {
-  // Somehow retrieve the virtual object model that will be related to the anchor.
-  let model = ...;
-  anchorToModelMap.put(anchor, model);
-}, (error) {
-  console.error(“Could not create arbitrary anchor: “ + error);
+let allAnchors = new Set();
+
+let referenceSpace = ...; // Reference space obtained from
+                          // XRSession.requestReferenceSpace(...).
+let anchorPose = new XRRigidTransform(...);
+// Create a free-floating anchor.
+frame.createAnchor(anchorPose, referenceSpace).then((anchor) => {
+  // Anchor created successfully - handle it.
+  allAnchors.add(anchor);
+
+  // For example, assign a model that will be placed relative to this anchor
+  // & add it to the scene. The location of the newly created anchor is not
+  // yet known but it should be by the time the application has a chance to
+  // render the object.
+}, (error) => {
+  console.error(“Could not create anchor: “ + error);
 });
 ```
 
 ### Removing anchors
 
 ```javascript
-for(var anchor of anchorToModelMap.keys()) {
+for(const anchor of allAnchors) {
   anchor.detach();
 }
-anchorToModelMap.clear();
+
+allAnchors.clear();
 ```
 
 ### Updating anchors
 
```

Comment:
There should be events for anchor updates and anchor deletion.  Since anchors may go many many frames without changing, or without being deleted, it is wasteful to check each frame.

Reply:
There are few reasons why I dislike events for anchor updates (deletions are another matter):
1. What does it mean that the anchor got updated? The anchor can remain unchanged, but that does not mean that the poses returned from `XFrame.getPose(anchorSpace, refSpace)` are not changing. In case of anchors, the applications will be most likely interested in their poses - because of that, I do not think we'd be able to provide an event that is useful to the application. This is also the reason why the code snippets in this PR do not rely on anchor's `lastUpdatedTime` attribute to perform the update.
2. The events that fire would not be useful to the application since anchor poses can only be obtained when the frame is active. We could make the events receive an activated frame, but that means the frame they get may (see 3.) contain stale data & they'd have to re-query the anchor poses in rAF callback.
3. In multi-process browsers, the data may be shipped from the device to the renderer process roughly once per frame - this is currently the approach that Chrome is taking. With this approach, all anchor update events could fire just prior to firing application's rAF callback, with up-to-date frame data. This means that they are effectively stealing time from rAF callback (it could already be fired), in a context that is not as capable as rAF callback (rendering can only occur in rAF callback).

As for anchor delete events, I'm leaning towards not adding them initially, since there is a way for the application to obtain the information about deleted anchors by inspecting the set of currently tracked anchors. The anchor removal events should be easy to add so we could revisit this later.

Reply:
I think all of that is fair.  Especially with (1) (lastUpdatedTime can serve as an indication of change) and (2) events are outside the rAF,they are less useful that they might be.